File: API/_archive/REST/golang/usecase1_crud_users.go
package main

import (
	"encoding/json"
	"net/http"
)

type User struct {
	ID   string `json:"id"`
	Name string `json:"name"`
}

func userHandler(w http.ResponseWriter, r *http.Request) {
	if r.Method == http.MethodGet {
		// Simulate DB Fetch
		user := User{ID: "123", Name: "Alice"}
		w.Header().Set("Content-Type", "application/json")
		json.NewEncoder(w).Encode(user)
	} else if r.Method == http.MethodPost {
		var newUser User
		json.NewDecoder(r.Body).Decode(&newUser)
		w.WriteHeader(http.StatusCreated) // 201
		json.NewEncoder(w).Encode(map[string]string{"status": "created"})
	}
}

func main() {
	http.HandleFunc("/users", userHandler)
	http.ListenAndServe(":8080", nil)
}

File: API/_archive/REST/python/usecase1_crud_users.py
from fastapi import FastAPI
from pydantic import BaseModel
import uvicorn

app = FastAPI()

class User(BaseModel):
    id: str
    name: str

@app.get("/users/{user_id}", response_model=User)
async def get_user(user_id: str):
    return User(id=user_id, name="Alice")

@app.post("/users", status_code=201)
async def create_user(user: User):
    return {"status": "created", "user": user}

if __name__ == "__main__":
    uvicorn.run(app, host="0.0.0.0", port=8000)
